libvast/src/system/catalog.cpp

```diff
@@ -491,62 +487,25 @@ catalog(catalog_actor::stateful_pointer<catalog_state> self,
     [=](atom::candidates, const vast::query_context& query_context)
       -> caf::result<catalog_result> {
       VAST_TRACE_SCOPE("{} {}", *self, VAST_ARG(query_context));
-      catalog_result expression_candidates;
-      std::vector<vast::uuid> ids_candidates;
       bool has_expression = query_context.expr != vast::expression{};
       bool has_ids = !query_context.ids.empty();
-      if (!has_expression && !has_ids)
+      if (has_ids)
+        return caf::make_error(ec::invalid_argument, "catalog expects queries "
+                                                     "not to have ids");
+      if (!has_expression)
```

Comment:
You can replace this with an assertion now.

Reply:
Discussed in Slack; comment is not applicable